### DISCUSSION TITLE
Update with null values issue

### DISCUSSION THREAD

**Discussion by pablojsx** [prisma/prisma#22331] [Q&A]
### Question

Hello, I'm in situation where I have a instance of a model, all good but there's a relation yet no set so the relation ID is set to null:

![Captura de pantalla 2023-12-10 121033](https://github.com/prisma/prisma/assets/40902487/c3027e84-887e-4a84-8aab-81477c5a93b7)

When I do a Prisma update, let's say I need to edit the deparment name, I get this error:

![Captura de pantalla 2023-12-10 121147](https://github.com/prisma/prisma/assets/40902487/ce8cc067-dc5e-4546-bc87-241cf9f752b6)

I also find counter intuitive that both ids are set to String?    @default(uuid()) but when creating a Department instance those are null.

Any help will be apreciated!

### How to reproduce (optional)

Create a prisma relation set to optional and set the ID of the relation to String? @default(uuid())

### Expected behavior (optional)

ids to be created and be strings until i create a relation

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

generator zod {
  provider                         = "zod-prisma-types"
  createInputTypes                 = false
  createOptionalDefaultValuesTypes = true
  output                           = "../src/server/zod/schemas"
}

model Department {
  departmentId      String     @id @default(uuid())
  departmentName    String     @unique @db.VarChar(255)
  description       String?    @db.Text
  jobs              Job[]
  employees         Employee[] @relation("departmentEmployees")
  manager           Employee?  @relation("employeeDepartmentManager", fields: [managerId], references: [employeeId]) // manager of the department
  managerId         String?    @default(uuid())
  storeDepartment   Store?     @relation(fields: [storeDepartmentId], references: [storeId])
  storeDepartmentId String?    @default(uuid())
  incompleteEntry   Boolean    @default(false)
  createdAt         DateTime   @default(now())
  updatedAt         DateTime   @updatedAt
}
```

  ```ts
updateDepartment: organizationProcedure
    .input(DepartmentOptionalDefaultsSchema)
    .mutation(async ({ input }) => {
      const updatedDepartment = await prisma.department.update({
        where: {
          departmentId: input.departmentId,
        },
        data: input,
      });
      return { updatedDepartment };
    }),

```

**Comment by ludralph** [comment]
Hi @pablojsx 👋 

Thank you for raising this question.

If I understand correctly, you are trying to perform an update and you are assigning null to a particular field and it throws the error message `expected string, received null`. In that case, Prisma expects, that you assign a valid string to the field in question.  My hunch is if for example, you have this query
```js
const res = await prisma.department.update({
  where:{
    departmentId: '42e03a67-e8f3-4a4d-bf49-fdccf659b137',
  },
  data:{
    departmentName: 'test2',
    description : 'test223',
    incompleteEntry: true,
    storeDepartment:{
      connect:{
        storeId: null
      }
    }
  }
});
```
Notice, how i'm assigning `null` to `storeId`, Prisma will throw an error because `storeId` is defined to be of type `String` in the `Store` model.

```prisma

model Store{
  storeId           String     @id @default(uuid())
  storeName         String     @unique @db.VarChar(255)
  description       String?    @db.Text
  departments       Department[]
  incompleteEntry   Boolean    @default(false)
  createdAt         DateTime   @default(now())
  updatedAt         DateTime   @updatedAt
}
```
Likewise if you assign say `null` to `employeeId` in `Employee` model, it would also throw an error because it is defined to be of type String. `managerId` and `storeDepartmentId` are foreign keys that depend on the values of `employeeId` and `storeId`. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
